src/kubernetes.cc

```diff
@@ -72,11 +70,20 @@ constexpr const char kKubernetesSchemaNameFormat[] =
 constexpr const char kClusterFullNameFormat[] =
     "//container.googleapis.com/projects/{{project_id}}/{{location_type}}/"
     "{{location}}/clusters/{{cluster_name}}";
-constexpr const char kNodeFullNameFormat[] =
-    "{{cluster_full_name}}/k8s/nodes/{{node_name}}";
-constexpr const char kPodFullNameFormat[] =
-    "{{cluster_full_name}}/k8s/namespaces/{{namespace_name}}/pods/{{pod_name}}";
+constexpr const char kK8sFullNameFormat[] =
+    "{{cluster_full_name}}/k8s/{{relative_link}}";
+
+constexpr const char kK8sCoreGroup[] = "core";
+constexpr const char kK8sCoreGroupResourceTypeFormat[] = "io.k8s.{{kind}}";
+constexpr const char kK8sNamedGroupsResourceTypeFormat[] =
+    "io.k8s.{{group_name}}.{{kind}}";
 
+// TODO: There seems to be a Kubernetes API bug with watch=true, so we
+// explicitly add "watch" to the path.
+constexpr const char kK8sCoreGroupWatchEndpointPathFormat[] =
+    "/api/{{version}}/watch/{{plural_kind}}";
+constexpr const char kK8sNamedGroupsWatchEndpointPathFormat[] =
+    "/apis/{{group_name}}/{{version}}/watch/{{plural_kind}}";
```

Comment:
It seems strange that you deconstruct the `apiVersion` only to re-construct it here again. Why not use `"/apis/{{apiVersion}}/watch/{{plural_kind}}"`?